File: FRI/Programing/Python/2Letnik/UV/DN06/naloga06.py
import pygit2
import gi

pygit2.require_version('Gtk', '3.0')
from pygit2.repository import Gtk


class Handler:
    def avtorClicked(self, *args):

        builder.get_object("label1").set_text("Jaz sem avtor")

    def izhodClicked(self, *args):
        Gtk.main_quit(args)

    def odpriClicked(self, *args):
        builder.get_object("filechooserdialog1").show_all()

    def prekliciClicked(self, *args):
        builder.get_object("filechooserdialog1").hide()

    def okClicked(self, *args):
        builder.get_object("label1").set_text(builder.get_object("filechooserdialog1").get_filename())
        builder.get_object("filechooserdialog1").hide()

    def pretvoriClicked(self, *args):
        tecaj = 1.0991
        vrednost = builder.get_object("spinbutton1").get_value_as_int()
        if (builder.get_object("comboboxtext1").get_active() == 0):
            vrednost *= tecaj
        else:
            vrednost = vrednost / tecaj
        if (builder.get_object("radiobutton1").get_active()): vrednost *= 0.98
        builder.get_object("entry1").set_text(str(vrednost))

    def izpisiClicked(self, *args):
        model = builder.get_object("comboboxtext1").get_model()
        size = 0
        iter = model.get_iter_first()
        curr = builder.get_object("comboboxtext1").get_active()
        while (iter):
            size = size + 1
            iter = model.iter_next(iter)
        for i in range(size):
            builder.get_object("comboboxtext1").set_active(i)
            print(i, builder.get_object("comboboxtext1").get_active_text())
        builder.get_object("comboboxtext1").set_active(curr)

    def dodajClicked(self, *args):
        builder.get_object("comboboxtext1").append_text(builder.get_object("entry1").get_text())


builder = Gtk.Builder()
builder.add_from_file("naloga06py.glade")
builder.connect_signals(Handler())
window = builder.get_object("window1")
window.show_all()

Gtk.main()

File: FRI/Programing/Python/2Letnik/UV/DN06/naloga06.c
#include <stdio.h>
#include <stdlib.h>
#include <gtk/gtk.h>

GtkBuilder *builder;
GtkWidget *window, *fc, *status, *comboB, *entry, *spinB, *rb;

//gboolean ime (GObject *obj GDKEvent event, gpointer userdata)

void avtorClicked(GObject *object, gpointer userdata){
  gtk_label_set_text(GTK_LABEL(status), "Jaz sem avtor!!!");
}
void odpriClicked(GObject *object, gpointer userdata){
  gtk_widget_show(fc);
}

void prekliciClicked(GObject *object, gpointer userdata){
  gtk_widget_hide(fc);
}

void okClicked(GObject *object, gpointer userdata){
  gtk_label_set_text(GTK_LABEL(status), 
    gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(fc)));
  gtk_widget_hide(fc);
}

void pretvoriClicked(GObject *object, gpointer userdata){
  double tecaj=1.0991;
  double vrednost = gtk_spin_button_get_value_as_int(GTK_SPIN_BUTTON(spinB));
  
  if (gtk_combo_box_get_active(GTK_COMBO_BOX(comboB))==0) vrednost*=tecaj;
  else vrednost/=tecaj;
  
  if (gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(rb))) vrednost*=0.98;
  
  gchar out[30];
  sprintf(out,"%10.3lf", vrednost);
  gtk_entry_set_text(GTK_ENTRY(entry),out);
}

gint elCnt(GtkTreeModel *model){
  gint size=0;
  GtkTreeIter iter;
  gboolean valid;
  valid=gtk_tree_model_get_iter_first(model, &iter);
  while (valid){
    size++;
    valid=gtk_tree_model_iter_next(model, &iter);
  }
  return size;
}

void izpisiClicked(GObject *object, gpointer userdata){
  int i,crt;
  int size = elCnt(gtk_combo_box_get_model(GTK_COMBO_BOX(comboB)));
  crt = gtk_combo_box_get_active(GTK_COMBO_BOX(comboB));
  for (i=0;i<size;i++){
    gtk_combo_box_set_active(GTK_COMBO_BOX(comboB), i);
    printf("%d %s\n", i, gtk_combo_box_text_get_active_text(GTK_COMBO_BOX_TEXT(comboB)));
  }
  gtk_combo_box_set_active(GTK_COMBO_BOX(comboB), crt);
}

void dodajClicked(GObject *object, gpointer userdata){
  gtk_combo_box_text_append_text(GTK_COMBO_BOX_TEXT(comboB), 
    gtk_entry_get_text(GTK_ENTRY(entry)));
}

int main(int argc, char *argv[]){
  gtk_init(&argc, &argv);
  builder=gtk_builder_new();
  gtk_builder_add_from_file(builder,"naloga06.glade", NULL);
  window = GTK_WIDGET(gtk_builder_get_object(builder,"window1"));
  fc = GTK_WIDGET(gtk_builder_get_object(builder,"filechooserdialog1"));
  status = GTK_WIDGET(gtk_builder_get_object(builder,"label1"));
  comboB = GTK_WIDGET(gtk_builder_get_object(builder,"comboboxtext1"));
  entry = GTK_WIDGET(gtk_builder_get_object(builder,"entry1"));
  spinB = GTK_WIDGET(gtk_builder_get_object(builder,"spinbutton1"));
  rb = GTK_WIDGET(gtk_builder_get_object(builder,"radiobutton1"));
  gtk_builder_connect_signals(builder,NULL);
  gtk_window_resize(GTK_WINDOW(window), 400, 250);
  gtk_widget_show(window);
  gtk_main();
  return 0;
}

